app/src/main/java/com/example/vmedvediev/ua21summerdancecamp/UI/bottomnavigation/events/EventsFragment.kt

```diff
@@ -48,27 +83,66 @@ class EventsFragment : Fragment(), TabLayout.OnTabSelectedListener {
     override fun onActivityCreated(savedInstanceState: Bundle?) {
         super.onActivityCreated(savedInstanceState)
         setupTabs()
-        Handler().postDelayed({onTabSelected(eventsTabLayout.getTabAt(0))}, 500)
+        Handler().postDelayed({ onTabSelected(eventsTabLayout.getTabAt(0)) }, 1)
         eventsTabLayout.addOnTabSelectedListener(this)
         setupRecycler()
 
-        viewModel.getEventsList().observe(this, Observer {
+        viewModel.getEventsLiveData().observe(this, Observer {
             eventsAdapter.notifyDataSetChanged()
         })
     }
 
+    override fun onTabReselected(tab: TabLayout.Tab?) {
+
+    }
+
+    override fun onTabUnselected(tab: TabLayout.Tab?) {
+
+    }
+
+    override fun onTabSelected(tab: TabLayout.Tab?) {
+        if (shouldTabBeSelected) {
+            listOfLastItemPositions = setupListOfLastItemPositions()
+            tab?.let {
+                viewModel.getDataBydDate((it.customView as TabCustomView).getDate())
+            }
+            eventsRecyclerView.scrollToPosition(0)
+            eventsAdapter.clearAndAddAll(viewModel.getEventsList())
+        } else {
+            tab?.let {
+                viewModel.getDataBydDate((it.customView as TabCustomView).getDate())
+                eventsAdapter.addAll(viewModel.getEventsList())
+            }
+        }
+    }
+
     private fun setupRecycler() {
         eventsRecyclerView.apply {
             setHasFixedSize(true)
-            layoutManager = LinearLayoutManager(activity)
+            layoutManager = linearLayoutManager
             adapter = eventsAdapter
+            addOnScrollListener(recyclerViewOnScrollListener)
         }
     }
 
     private fun setupTabs() {
-        for (i in 0..7) {
-            eventsTabLayout.addTab(eventsTabLayout.newTab())
-            eventsTabLayout.getTabAt(i)?.customView = TabCustomView(activity as AppCompatActivity, namesOfDaysArray[i], numberOfDaysArray[i])
+        for (i in IntRange(INDEX_OF_FIRST_TAB, INDEX_OF_LAST_TAB)) {
+            eventsTabLayout.apply {
+                addTab(eventsTabLayout.newTab())
+                getTabAt(i)?.customView = TabCustomView(activity as AppCompatActivity, namesOfDaysArray[i], numberOfDaysArray[i])
+            }
+        }
+    }
+
+    private fun setupListOfLastItemPositions() : ArrayList<Int> {
+        val listOfLastItemPositions = ArrayList<Int>()
+        var tempLastPosition = -1
+        for (i in IntRange(INDEX_OF_FIRST_TAB, INDEX_OF_LAST_TAB)) {
```

Comment:
INDEX_OF_FIRST_TAB = 0 (it's obvious)
INDEX_OF_LAST_TAB = tabLayout.size
So, probably, you don't need these constants...